### PR TITLE
Feature/JS-3528: Featured relation tag and status editing

### PR DESCRIPTION
**Pull request by mhlv**
---

- [x] I understand that contributing to this repository will require me to agree with the [CLA](https://github.com/anyproto/.github/blob/main/docs/CLA.md)

---

### What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI

### Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

### Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 [tech-docs](https://github.com/anyproto/tech-docs)
- [ ] 🙅 no documentation needed

### CODE REVIEW

File: src/scss/theme/dark/menu.scss [hunk 1]
```diff
@@ -212,6 +212,23 @@
 
     .menu.menuDataviewOptionEdit {
 		.item:hover::before { background: $colorShapeHighlightMedium; }
+
+        .section.colorPicker {
+            .items {
+                .item.hover, .item:hover { box-shadow: 0 0 0 1px $colorBgPrimary, 0 0 0 3px $colorShapePrimary; }
```
Comment (ra3orblade) [inline src/scss/theme/dark/menu.scss:218]: add px everywhere

File: src/ts/component/block/featured.tsx [hunk 2]
```diff
@@ -713,6 +719,41 @@ const BlockFeatured = observer(class BlockFeatured extends React.Component<Props
 		});
 	};
 
+	onTagOrStatus (e: React.MouseEvent, relationKey: string) {
```
Comment (ra3orblade) [inline src/ts/component/block/featured.tsx:722]: change to onCellSelect

File: src/ts/component/menu/dataview/option/list.tsx [hunk 3]
```diff
@@ -317,7 +339,7 @@ const MenuOptionList = observer(class MenuOptionList extends React.Component<I.M
 	getItems (): any[] {
 		const { param } = this.props;
 		const { data } = param;
-		const { canAdd, filterMapper } = data;
+		const { canAdd, filterMapper, keepSelected } = data;
```
Comment (ra3orblade) [inline src/ts/component/menu/dataview/option/list.tsx:342]: Remove this logic, make keepSelected by default, we have discussed it.

File: src/ts/component/menu/dataview/option/list.tsx [hunk 4]
```diff
@@ -234,6 +238,12 @@ const MenuOptionList = observer(class MenuOptionList extends React.Component<I.M
 		const { onChange, maxCount } = data;
 
 		let value = Relation.getArrayValue(data.value);
+
+		if (value.includes(id)) {
+			this.onValueRemove(id);
```
Comment (ra3orblade) [inline src/ts/component/menu/dataview/option/list.tsx:243]: this is not correct, onChange handler is not called.

File: src/ts/component/menu/dataview/option/list.tsx [hunk 5]
```diff
@@ -184,6 +187,7 @@ const MenuOptionList = observer(class MenuOptionList extends React.Component<I.M
 	unbind () {
 		$(window).off('keydown.menu');
 		$(`#${this.props.getId()}`).off('click');
+		menuStore.close('dataviewOptionEdit');
```
Comment (ra3orblade) [inline src/ts/component/menu/dataview/option/list.tsx:190]: this is not correct, unbind should not close menu

File: src/scss/form/filter.scss [hunk 6]
```diff
@@ -21,3 +21,12 @@
 .filter.active { 
 	.icon.clear { opacity: 1; } 
 }
+.filter.v2 {
```
Comment (ra3orblade) [inline src/scss/form/filter.scss:24]: I do not like v2 class name

File: src/ts/component/block/featured.tsx [hunk 7]
```diff
@@ -608,6 +609,11 @@ const BlockFeatured = observer(class BlockFeatured extends React.Component<Props
 			return;
 		};
 
+		if (relation.format == I.RelationType.Tag || relation.format == I.RelationType.Status) {
+			this.onTagOrStatus(e, relationKey);
```
Comment (ra3orblade) [inline src/ts/component/block/featured.tsx:613]: change to [].includes()

File: src/ts/component/block/featured.tsx [hunk 8]
```diff
@@ -5,6 +5,7 @@ import { ObjectType, Cell } from 'Component';
 import { I, C, UtilData, UtilCommon, UtilObject, UtilDate, Preview, focus, analytics, Relation, Onboarding, history as historyPopup, keyboard, translate } from 'Lib';
 import { blockStore, detailStore, dbStore, menuStore, commonStore } from 'Store';
 import Constant from 'json/constant.json';
+import { observable } from 'mobx';
```
Comment (ra3orblade) [inline src/ts/component/block/featured.tsx:8]: Move import from mobx to library imports, they usually are at the start of the document

File: src/ts/component/menu/dataview/option/list.tsx (hunk 4 above)
Comment (mhlv) [reply src/ts/component/menu/dataview/option/list.tsx:243]: it's called from onValueRemove method. I split this logic to increase readability, can combine them back of course.

File: src/scss/form/filter.scss (hunk 6 above)
Comment (mhlv) [reply src/scss/form/filter.scss:24]: it's temporary solution until transition is done

File: src/ts/component/menu/dataview/option/list.tsx (hunk 4 above)
Comment (ra3orblade) [reply src/ts/component/menu/dataview/option/list.tsx:242]: looks ugly and not readable

File: src/ts/component/block/featured.tsx [hunk 9]
```diff
@@ -713,6 +719,43 @@ const BlockFeatured = observer(class BlockFeatured extends React.Component<Props
 		});
 	};
 
+	onCellSelect (e: React.MouseEvent, relationKey: string) {
+		const { rootId, block } = this.props;
+		const storeId = this.getStoreId();
+		const object = detailStore.get(rootId, storeId, [ relationKey ]);
+		const relation = dbStore.getRelationByKey(relationKey);
+		const value = object[relationKey];
```
Comment (ra3orblade) [inline src/ts/component/block/featured.tsx:727]: value = Relation.getArrayValue(object[relationKey])

File: src/ts/component/menu/dataview/option/edit.tsx [hunk 10]
```diff
@@ -20,14 +20,23 @@ const MenuOptionEdit = observer(class MenuOptionEdit extends React.Component<I.M
...
+				onClick={e => this.onClick(e, item)}
+				onMouseEnter={e => this.onMouseEnter(e, item)}
+			>
+				{this.color == item.value ? <Icon className="tick" /> : ''}
+			</div>
+		);
+
 		const Section = (item: any) => (
-			<div className="section">
+			<div className={[ 'section', item.className ? item.className : '' ].join(' ')}>
```
Comment (ra3orblade) [inline src/ts/component/menu/dataview/option/edit.tsx:35]: Wrap item.className ? item.className : '' in parenthesis

**Pull request closed without merging**